bundles/automation/org.eclipse.smarthome.automation.module.core/src/main/java/org/eclipse/smarthome/automation/module/core/handler/ChannelPayloadTriggerHandler.java

```diff
@@ -0,0 +1,55 @@
+/**
+ * Copyright (c) 1997, 2015 by ProSyst Software GmbH and others.
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v1.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-v10.html
+ */
+package org.eclipse.smarthome.automation.module.core.handler;
+
+import org.apache.commons.lang.StringUtils;
+import org.eclipse.smarthome.automation.Trigger;
+import org.eclipse.smarthome.core.events.Event;
+import org.eclipse.smarthome.core.thing.events.ChannelTriggeredEvent;
+import org.osgi.framework.BundleContext;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * This is an ModuleHandler implementation for Trigger Channel with specific events
+ *
+ * @author Stefan Triller - Initial contribution
+ *
+ */
+public class ChannelPayloadTriggerHandler extends GenericEventTriggerHandler {
```

Comment:
Please don't extend `GenericEventTriggerHandler`, but do it similar to `ItemStateTriggerHandler`.
Rename to `ChannelEventTriggerHandler`.